flatten_dict/flatten_dict.py

```diff
@@ -41,9 +41,9 @@ def flatten(d, reducer='tuple', inverse=False):
     flat_dict = {}
 
     def _flatten(d, parent=None):
-        for key, value in six.viewitems(d):
+        for key, value in (six.viewitems(d) if isinstance(d, dict) else enumerate(d)):
             flat_key = reducer(parent, key)
-            if isinstance(value, Mapping):
+            if isinstance(value, (Mapping, list)):
```

Comment:
`list` -> `collections.Iterable`

Reply:
Oh, but `str` is also `Iterable`. Let me think about this later.

Reply:
OK. I think it would be better to let user define the type to do the flattening.